pkg/devfile/adapters/common/command.go

```diff
@@ -164,6 +179,38 @@ func validateCommand(data data.DevfileData, command common.DevfileCommand) (err
 	return
 }
 
+// validateCompositeCommand checks that the specified composite command is valid
+func validateCompositeCommand(data data.DevfileData, compositeCommand *common.Composite) error {
+	if compositeCommand.Group.Kind == common.RunCommandGroupType {
+		return fmt.Errorf("composite commands of run Kind are not supported currently")
+	}
+
+	commandsMap := GetCommandsMap(data.GetCommands())
+
+	// Loop over the commands and validate that each command points to a command that's in the devfile
+	for _, command := range compositeCommand.Commands {
+		if command == compositeCommand.Id {
+			return fmt.Errorf("the composite command %q cannot reference itself", compositeCommand.Id)
+		}
+
+		_, ok := commandsMap[command]
+		if !ok {
+			return fmt.Errorf("the command %q does not exist in the devfile", command)
+		}
+	}
+	return nil
+}
+
+// hasCommand returns true if the devfile contains the specified command
+func hasCommand(data data.DevfileData, command string) bool {
```

Comment:
I assume you were using this before switching to map, looking at the previous review comments?
Since you left this here, there is a chance someone else may use it instead of the map. Can we move the map logic to this function, so that anyone can use it?

Reply:
Done